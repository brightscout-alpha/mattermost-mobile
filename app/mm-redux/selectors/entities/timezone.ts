// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {GlobalState} from '@mm-redux/types/store';
import {getCurrentUserId} from '@mm-redux/selectors/entities/common';
import {createSelector} from 'reselect';

export function getUserTimezone(state: GlobalState, id: string) {
    const profile = state.entities.users.profiles[id];

    if (profile && profile.timezone) {
        return {
            ...profile.timezone,
            useAutomaticTimezone: profile.timezone.useAutomaticTimezone === 'true',
        };
    }

    return {
        useAutomaticTimezone: true,
        automaticTimezone: '',
        manualTimezone: '',
    };
}

export function isTimezoneEnabled(state: GlobalState) {
    const {config} = state.entities.general;
    return config.ExperimentalTimezone === 'true';
}

export const getCurrentUserTimezone = createSelector(
    getCurrentUserId,
    isTimezoneEnabled,
    (state: GlobalState) => (userId: string) => getUserTimezone(state, userId),
    (userId, enabledTimezone, getTimezone) => {
        let timezone;
        if (enabledTimezone) {
            const userTimezone = getTimezone(userId);
            timezone = userTimezone.useAutomaticTimezone ? userTimezone.automaticTimezone : userTimezone.manualTimezone;
        }

        return timezone;
    },
);
