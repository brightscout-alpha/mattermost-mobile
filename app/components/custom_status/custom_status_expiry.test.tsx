// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import * as PreferenceSelectors from '@mm-redux/selectors/entities/preferences';
import {renderWithReduxIntl} from 'test/testing_library';
import CustomStatusExpiry from './custom_status_expiry';
import Preferences from '@mm-redux/constants/preferences';

jest.mock('@mm-redux/selectors/entities/preferences');

describe('components/custom_status/custom_status_emoji', () => {
    const date = '2200-04-13T18:09:12.451Z';

    const baseProps = {
        theme: Preferences.THEMES.default,
        time: new Date(date),
    };

    (PreferenceSelectors.getBool as jest.Mock).mockReturnValue(false);
    it('should match snapshot', () => {
        const wrapper = renderWithReduxIntl(
            <CustomStatusExpiry
                {...baseProps}
            />,
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
