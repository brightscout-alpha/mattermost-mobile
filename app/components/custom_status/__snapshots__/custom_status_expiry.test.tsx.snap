// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/custom_status/custom_status_expiry should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#3d3c40",
        "fontSize": 15,
      },
      undefined,
    ]
  }
>
  <Text>
    Apr 13, 2200
  </Text>
</Text>
`;
